includes/Validation/RequestValidationHelper.php

```diff
@@ -193,44 +188,69 @@ public function validateOther()
             $errorList[ValidationError::BANNED] = new ValidationError(ValidationError::BANNED_TOR);
         }
 
-        // IP banned
-        $ban = $this->banHelper->ipIsBanned($trustedIp);
-        if ($ban != false) {
+        // Bans
+        if ($this->banHelper->isBlockBanned($request)) {
             $errorList[ValidationError::BANNED] = new ValidationError(ValidationError::BANNED);
         }
 
-        // WARNINGS
+        return $errorList;
+    }
 
+    public function postSaveValidations(Request $request)
+    {
         // Antispoof check
-        $this->checkAntiSpoof();
+        $this->checkAntiSpoof($request);
 
         // Blacklist check
-        $this->checkTitleBlacklist();
+        $this->checkTitleBlacklist($request);
 
-        return $errorList;
+        $bans = $this->banHelper->getBans($request);
```

Comment:
Just wishing to verify that this goes through only the active bans without dumpster diving the code. Resolve if it does.

Reply:
`BanHelper` gets a list of bans which apply to supplied `Request` object.

The `getBans(Request)` call is a caching wrapper around `getBansForRequestFromDatabase(Request)`, which in turn runs a database query to find any matching bans. For most attributes, such as names, this is a simple `coalesce(:name rlike name, true)` - in other words, match the requested name to a regex stored in the ban table, and if the result is `null`, return `true`. IP ranges are a little more complex, as it involves a join to the new `netmask` table to check ranges, but it's essentially doing the same thing. We also filter where `active = 1` and `duration` isn't in the past. See [L157 of `BanHelper`](https://github.com/enwikipedia-acc/waca/blob/1cbaa6f61f118374c60d248067a9a3e7afa8c0c2/includes/Helpers/BanHelper.php#L157-L174) for the full SQL query.

We then take that list of applicable bans, and loop through them, deciding what to do based on the type of ban. If it's a "do nothing ban", we do nothing. If it's a "defer" ban, we defer it to the relevant queue. If it's a drop ban, we do the relevant drop logic with relevant log entries so it is traceable that a request was created and silently dropped according to the user. Note that "block"-type bans are handled as a pre-save validation, not as a post-save validation, so we don't need to worry about them at this stage.